scripts/west_commands/build.py

```diff
@@ -78,20 +87,17 @@ def do_add_parser(self, parser_adder):
             self.name,
             help=self.help,
             formatter_class=argparse.RawDescriptionHelpFormatter,
-            description=self.description)
+            description=self.description,
+            usage='''west build [-h] [-b BOARD] [-d BUILD_DIR]
+                  [-t TARGET] [-c] [-f] [source_dir]
+                  -- [cmake_opt [cmake_opt ...]]''')
 
         # Remember to update scripts/west-completion.bash if you add or remove
         # flags
 
         parser.add_argument('-b', '--board',
                             help='''Board to build for (must be given for the
                             first build, can be omitted later)''')
-        parser.add_argument('-s', '--source-dir',
```

Comment:
Thinking about this, are you opposed to keeping this option and its behavior, but just making its help `argparse.SUPPRESS` (or marking it deprecated)?

People are used to how this works and I'm not sure breaking scripts that use this is a good idea, though I don't oppose the new method of making it a positional argument.

Reply:
Not opposed, I will add this as a "hidden" option.